docs/architecture/006-authentication.md

```diff
@@ -0,0 +1,183 @@
+     ```
+     {"kty":"oct",
+      "k":"AyM1SysPpbyDfgZld3umj1qzKObwVMkoqQ-EstJQLr_T-1qS0gZH75
+           aKtMN3Yj0iPS4hcgUuTwjAzZr1Z9CAow"
+     }
+     ```
+     (which is not included in the JWS message, but could be linked with `kid` parameter)
+     the final message is:
+     ```
+     eyJ0eXAiOiJKV1QiLA0KICJhbGciOiJIUzI1NiJ9
+     .
+     eyJpc3MiOiJqb2UiLA0KICJleHAiOjEzMDA4MTkzODAsDQogImh0dHA6Ly9leGFt
+     cGxlLmNvbS9pc19yb290Ijp0cnVlfQ
+     .
+     dBjftJeZ4CVP-mB92K27uhbUJU1p1r_wW1gFWFOEjXk
+     ```
+
+JWE: JSON Web Encryption, RFC 7516
+     A standard format to encrypt JSON data.
+
+     Consists of two parts:
+     JWE Protected Header and the payload
+
+     Human readable header
+     ```
+     {"alg":"RSA-OAEP","enc":"A256GCM"}
+     ```
+     and payload of "The true sign of intelligence is not knowledge but imagination."
+     could be encoded (the initialization vector is random) as:
+     ```
+     eyJhbGciOiJSU0EtT0FFUCIsImVuYyI6IkEyNTZHQ00ifQ.
+     OKOawDo13gRp2ojaHV7LFpZcgV7T6DVZKTyKOMTYUmKoTCVJRgckCL9kiMT03JGe
+     ipsEdY3mx_etLbbWSrFr05kLzcSr4qKAq7YN7e9jwQRb23nfa6c9d-StnImGyFDb
+     Sv04uVuxIp5Zms1gNxKKK2Da14B8S4rzVRltdYwam_lDp5XnZAYpQdb76FdIKLaV
+     mqgfwX7XWRxv2322i-vDxRfqNzo_tETKzpVLzfiwQyeyPGLBIO56YJ7eObdv0je8
+     1860ppamavo35UgoRdbYaBcoh9QcfylQr66oc6vFWXRcZ_ZT2LawVCWTIy3brGPi
+     6UklfCpIMfIjf7iGdXKHzg.
+     48V1_ALb6US04U3b.
+     5eym8TW_c8SuK0ltJ3rpYIzOeDQz7TALvtu6UG9oMo4vpzs9tX_EFShS8iB7j6ji
+     SdiwkIr3ajwQzaBtQD_A.
+     XFBoMYUZodetZdvTiFvSkQ
+     ```
+     (the used public key isn't included or even specified in the message).
+
+JWK: JSON Web Key, RFC 7517
+     A standard format to represent cryptographic keys as JSON. For example a symmetric key:
+     ```
+     {"kty":"oct",
+      "k":"AyM1SysPpbyDfgZld3umj1qzKObwVMkoqQ-EstJQLr_T-1qS0gZH75
+           aKtMN3Yj0iPS4hcgUuTwjAzZr1Z9CAow"
+     }
+     ```
+
+JWT: JSON Web Token, RFC 7519; JSON based, Base64url encoded format for transferring claims.
+     Data can be transferred as unsecured, or in JWS or JWE payload
+     (note that OIDC allows only signed JWT, with or without encryption).
+
+OAuth2: RFC 6749; authorization protocol based on browser redirects and JSON, supersedes OAuth1
+OIDC: OpenID Connect; authentication protocol on top of OAuth2 and signed JWT, supersedes OpenId 2.0
+OIDC client: a system that requires authentication services — in this context REMS is a client, while the actual person is user and service doing the authentication is identity provider
+SAML: Security Assertion Markup Language; XML based protocol for authentication and authorization
+Shibboleth: Single Sign-On system based on SAML
+Elixir: EU level life sciences consortium
+Elixir AAI proxy: Autehntication and Authorisation Infrastructure; Elixir's authentication system that aggregates different ways for users to log in, and works as a SAML and OIDC identity provider for other systems, like REMS
```

Comment:
I suppose each of these lines is meant to be its own paragraph. Now markdown clumps them all together in one paragraph.

Reply:
Also those parts of the doc which are indented 4 spaces are rendered as a big code block (except for the first line which is clumped in the same paragraph with the previous line), which is probably not what was intended. Check the preview.